lib/modules/datasource/unity3d/index.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { getPkgReleases } from '..';
+import { Fixtures } from '../../../../test/fixtures';
+import * as httpMock from '../../../../test/http-mock';
+import { Unity3dDatasource } from '.';
+
+const mockRSSFeeds = (streams: { [keys: string]: string }) => {
+  Object.entries(streams).map(([stream, url]) => {
+    const content = Fixtures.get(stream + '.xml');
+
+    const uri = new URL(url);
+
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, content);
+  });
+};
+
+describe('modules/datasource/unity3d/index', () => {
+  it('handle 500 response', async () => {
+    const uri = new URL(Unity3dDatasource.streams.stable);
+    httpMock.scope(uri.origin).get(uri.pathname).reply(500, '500');
+
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    const response = await getPkgReleases({
+      datasource: Unity3dDatasource.id,
+      packageName: 'm_EditorVersion',
+      registryUrls: [Unity3dDatasource.streams.stable],
+    });
+
+    expect(response).toBeNull();
+  });
+
+  it('handle 200 with no XML', async () => {
+    const uri = new URL(Unity3dDatasource.streams.stable);
+    httpMock.scope(uri.origin).get(uri.pathname).reply(200, 'not xml');
+
+    const qualifyingStreams = { ...Unity3dDatasource.streams };
+    delete qualifyingStreams.beta;
+    const response = await getPkgReleases({
+      datasource: Unity3dDatasource.id,
+      packageName: 'm_EditorVersion',
+      registryUrls: [Unity3dDatasource.streams.stable],
+    });
+
+    expect(response).toBeNull();
+  });
+
+  it('handles missing title element', async () => {
+    const content = Fixtures.get('no_title.xml');
```

Comment:
Load the fixtures on the module scope, so we do not load them into memory per test

Reply:
Resolved inside 246ceae75585c009405b76ffbbc80c1a28e9132c.